### PR TITLE
feat(infra): allow usage of existing VPC coming with AWS account

### PR DESCRIPTION
**Pull request by imincik**
This change is introducing following features:

* possibility to use existing VPC provided by/coming with AWS account
* or in case we create VPC by our networking stack, it is created with limited networking resources (NAT GW, EIPs) if deployment environment is not `prod`. This will reduce cost of networking resources for test and CI deployments and will
reduce usage of EIPs (which is limited to 5 EIPs/AWS account).

Closes: #134

### CODE REVIEW

**Comment by imincik** [comment]
This PR might need manual dropping of CI networking stack to succeed.

File: datalake/networking_stack.py [hunk 1]
```diff
@@ -32,5 +32,6 @@ def __init__(self, scope: core.Construct, stack_id: str, **kwargs) -> None:
                     reserved=True,
                 ),
             ],
+            max_azs=99 if deploy_env == "prod" else 1,
```
Comment (l0b0) [inline datalake/networking_stack.py:35]: We should pull out an [enum](https://docs.python.org/3/library/enum.html) for the environment names sooner rather than later. Up to you whether you want to create a separate PR for that.

**Comment by l0b0** [comment]
> This PR might need manual dropping of CI networking stack to succeed.

Why?

File: datalake/networking_stack.py (hunk 1 above)
Comment (imincik) [reply datalake/networking_stack.py:35]: I would rather add enum in other PR. It needs to be used at multiple places.

Comment (imincik) [reply datalake/networking_stack.py:35]: > Also, I expect max_azs=None is more appropriate than 99.

From CDK docs describing `max_azs` parameter looks like using value such as `99` is recommended.

"""
To use “all AZs” available to your account, use a high number (such as 99).
"""

Comment (imincik) [reply datalake/networking_stack.py:35]: Docs is not saying what's going to happen if I set None.

**Comment by imincik** [comment]
@l0b0 , this PR was significantly changed.

File: app.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
     "networking",
     stack_name=f"geospatial-data-lake-networking-{ENV}",
     env={"region": os.environ["CDK_DEFAULT_REGION"], "account": os.environ["CDK_DEFAULT_ACCOUNT"]},
+    deploy_env=ENV,
+    use_existing_vpc=True,  # use already existing VPC, pre-configured in AWS account
```
Comment (l0b0) [inline app.py:27]: As the code stands it's not possible to call this with `False` without changing the code. I'd probably do something like `use_existing_vpc = str2bool(os.environ.get("USE_EXISTING_VPC"))` where

```
def str2bool(value: str) -> bool:
    if value == "True":
        return True
    if value == "False":
        return False
    raise ValueError(f"Not a valid boolean: '{value}'")
```

Up to you if you want to do that in a separate PR.

Comment (imincik) [reply app.py:27]: Yes , this is something which I am aware of. I was thinking about env variable as well.

Comment (imincik) [reply app.py]: Should be done in d1e2abaeab5a4c9946386678b3b06a97a246ffbe

File: README.md [hunk 3]
```diff
@@ -93,10 +93,16 @@ Re-run `. .venv/bin/activate` in each shell.
     ```bash
     aws-azure-login --profile=<AWS-PROFILE-NAME>
     ```
+1. Environment variables
+* **DEPLOY_ENV:** set deployment environment. Recommended values: prod, nonprod, ci, dev or any
+    string without spaces
+* **DATALAKE_USE_EXISTING_VPC:** determine if networking stack will use existing VPC in target AWS
```
Comment (l0b0) [inline README.md:99]: We should document the default here.

File: app.py [hunk 4]
```diff
@@ -16,13 +16,23 @@
...
+
 app = core.App()
 
 networking = NetworkingStack(
     app,
     "networking",
     stack_name=f"geospatial-data-lake-networking-{ENV}",
     env={"region": os.environ["CDK_DEFAULT_REGION"], "account": os.environ["CDK_DEFAULT_ACCOUNT"]},
+    deploy_env=ENV,
+    use_existing_vpc=str2bool(os.environ.get("DATALAKE_USE_EXISTING_VPC", "true")),
```
Comment (l0b0) [inline app.py:35]: Do we want to use `True` as the default? I would've thought `False` would be safer.

**Review by l0b0 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (imincik) [reply README.md:99]: It is documented now and `DATALAKE_USE_EXISTING_VPC` default is `false`

**Review by l0b0 — APPROVED** [review]
